src/game/hud.cpp

```diff
@@ -1758,49 +1771,58 @@ namespace hud
 
     void drawconsole(int type, int w, int h, int x, int y, int s, float fade)
     {
-        static vector<int> refs; refs.setsize(0);
+        static vector<int> refs; 
+        refs.setsize(0);
+        
         bool full = fullconsole || commandmillis > 0;
         int tz = 0;
         pushfont("console");
-        if(type >= 2)
+        if (type >= 2)
         {
-            int numl = chatconsize, numo = chatconsize+chatconoverflow;
-            loopvj(conlines) if(conlines[j].type >= CON_CHAT)
+            int numl = chatconsize, numo = chatconsize + chatconoverflow;
+
+            for (auto j = chatpos; j < chatpos + min(numo, chatlines.length() - chatpos); j++)
             {
-                int len = !full && conlines[j].type > CON_CHAT ? chatcontime/2 : chatcontime;
-                if(full || totalmillis-conlines[j].reftime <= len+chatconfade)
+                int len = !full && chatlines[j].type > CON_CHAT ? chatcontime / 2 : chatcontime;
+                if (full || totalmillis - chatlines[j].reftime <= len + chatconfade)
```

Comment:
I would also negate this and use continue.

Reply:
same as before, not here to improve the existing code